src/lpython/parser/semantics.h

```diff
@@ -258,8 +258,29 @@ static inline char *extract_type_comment(Allocator &al, LFortran::Str &s) {
     str.erase(str.find_last_not_of(' ') + 1); // trim right spaces
     str.erase(0, str.find_first_not_of(' ')); // trim left spaces
 
-    s.from_str_view(str);
-    return s.c_str(al);
+    size_t i = 0;
+    bool flag = true;
+    std::string kw{"ignore"};
+    while (i < 6) {
+        if(str[i] == kw[i]) {
+            i++;
+        } else {
+            flag = false;
+            break;
+        }
+    }
+
+    if (flag) {
+        pos = 6;
+        str = str.substr(pos, str.size());
+        s.from_str_view(str);
+        p.type_ignore.push_back(p.m_a, down_cast<type_ignore_t>(
+            make_TypeIgnore_t(p.m_a, loc, 0, s.c_str(p.m_a))));
```

Comment:
cc @certik.
Do we store the Line number information in the `tokenizer`? As I can see only the start and end of the token information in the `Location`.
Here we have to pass Line number instead of `0`

Reply:
We can compute the corresponding line using the LocationManager class from the `loc` variable. For now just set 0, we can figure out later how to do this -- either after AST is constructed, or (probably better), we pass in the location manager into the `Parser` class and then access it using `p.m_location_manager` above, and convert the linear location `start` into line+column.

Reply:
We do not store the `lineno` members, we use our location information instead. I think this `lineno` is there specified manually (instead of using the `attributes` part in ASDL). I think we can just remove it from `Python.asdl`, and then we don't have to set it to anything. The full location is stored there using the `loc` approach.

Reply:
I left `TypeIgnore` unchanged in the `Python.asdl`, because it causes the Old Parser to fail.

Reply:
Yes, the old parser would also need to be modified.